sound/soc/sof/intel/hda-dai.c

```diff
@@ -286,6 +286,37 @@ static int hda_link_pcm_prepare(struct snd_pcm_substream *substream,
 				  dai);
 }
 
+static int hda_link_dai_config_pause_ipc(struct snd_soc_dapm_widget *w)
+{
+	struct snd_sof_widget *swidget = w->dobj.private;
+	struct snd_soc_component *component = swidget->scomp;
+	struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(component);
+	struct sof_ipc_dai_config *config;
+	struct snd_sof_dai *sof_dai;
+	struct sof_ipc_reply reply;
+	int ret;
+
+	sof_dai = swidget->private;
+
+	if (!sof_dai || !sof_dai->dai_config) {
+		dev_err(sdev->dev, "No config for DAI %s\n", w->name);
+		return -EINVAL;
+	}
+
+	config = &sof_dai->dai_config[sof_dai->current_config];
+
+	/* set PAUSE flag */
+	config->flags = FIELD_PREP(SOF_DAI_CONFIG_FLAGS_MASK, SOF_DAI_CONFIG_FLAGS_PAUSE);
+
+	/* send DAI_CONFIG IPC */
+	ret = sof_ipc_tx_message(sdev->ipc, config->hdr.cmd, config, config->hdr.size,
+				  &reply, sizeof(reply));
+	if (ret < 0)
+		dev_err(sdev->dev, "failed setting DAI config for %s\n", w->name);
+
+	return ret;
+}
+
```

Comment:
Suggested edit to the commit message

For HDA DAI's, the recommended programming sequence during PAUSE_PUSH is to clear the RUN bit, then send an IPC to perform additional DMA-related changes (clear DGCS.GEN bin).

This second part is handled by reusing the DAI_CONFIG IPC with the SOF_DAI_CONFIG_FLAGS_PAUSE_PUSH flag set

Reply:
also wondering if we can reuse the existing widget_update function, this looks like a copy-paste, no?